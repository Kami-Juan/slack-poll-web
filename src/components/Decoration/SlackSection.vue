<template>
  <div class="slack-section">
    <div class="separator"></div>
    <div class="slack-slot">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SlackSection'
}
</script>

<style>
.slack-section {
  text-align: left;
  display: flex;
}

.separator {
  width: 4px;
  align-items: stretch;
  background-color: var(--yellowme-color);
  border-radius: 2px;
}

.slack-slot {
  padding-top: 3px;
  padding-bottom: 4px;
  margin-left: 13px;
}
</style>