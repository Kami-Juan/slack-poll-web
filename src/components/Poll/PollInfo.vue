<template>
  <div id="poll-info">
    <div id="bot-avatar"></div>
    <div id="bot-message">
      <h4 id="bot-name">Yellow Poll</h4>
      <slack-section>
        <div id="info-group">
          <h4 id="poll-title">{{ title }}</h4>
          <div>By: You</div>
          <div>
            Mode:
            <a id="poll-type" @click="togglePollType">{{ multiple ? 'Multiple' : 'Single' }}</a>
          </div>
        </div>
      </slack-section>
      <div id="options-slot">
        <slot></slot>
      </div>
    </div>
  </div>
</template>

<script>
import SlackSection from '../Decoration/SlackSection'

export default {
  name: 'PollInfo',
  components: {
    SlackSection
  },
  props: ['title'],
  methods: {
    togglePollType() {
      this.multiple = !this.multiple
      this.$emit('typeChanged', this.multiple)
      this.$emit('updated')
    }
  },
  data() {
    return {
      multiple: false
    }
  }
}
</script>

<style>
#poll-info {
  margin-bottom: 10px;
  align-items: center;
  display: flex;
  flex-direction: row;
  align-items: start;
}

#poll-title {
  font-family: LatoBold;
  font-size: 19px;
  font-weight: bold;
  margin-top: 0;
  margin-bottom: 5px;
}

#poll-type {
  font-family: LatoBold;
  font-weight: bold;
  text-decoration: underline;
}

#bot-avatar {
  width: 40px;
  height: 40px;
  border-radius: 3px;
  background-color: var(--yellowme-color);
  margin-right: 10px;
}

#bot-name {
  font-family: LatoBold;
  font-size: 19px;
  font-weight: bold;
  margin-top: 0;
  margin-bottom: 2px;
  text-align: left;
}

#options-slot {
  margin-top: 10px;
}
</style>