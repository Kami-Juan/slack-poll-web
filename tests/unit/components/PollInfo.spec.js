import { mount } from '@vue/test-utils'

import PollInfo from './../../../src/components/Poll/PollInfo'
import SlackSection from './../../../src/components/Decoration/SlackSection'

describe('test PollInfo.vue', () => {
  let cmp

  beforeEach(() => {
    const slackSectionWrapper = {
      render(h) {
        return h(SlackSection)
      }
    }

    cmp = mount(PollInfo, {
      propsData: {
        title: 'I"m title',
        multiple: false
      },
      components: {
        SlackSection
      },
      slots: {
        default: slackSectionWrapper
      }
    })
  })

  it('should be render', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  it('the bot name has the correct text', () => {
    const botNameText = cmp.find('#bot-name').text()
    expect(botNameText).toBe('Yellow Poll')
  })

  it('the props has the correct validations', () => {

    const props = cmp.vm.$options.props

    expect(props.title.required).toBeTruthy()
    expect(props.title.type).toBe(String)

    expect(props.multiple.type).toBe(Boolean)
  })

  it('the poll title has the correct text', () => {
    cmp.setProps({title: 'I"m a poll title'})

    const pollTitleText = cmp.find('#poll-title').text()
    expect(pollTitleText).toBe('I"m a poll title')
  })

  it('the poll username has the correct text', () => {
    const pollUsernameText = cmp.find('#poll-username').text()
    expect(pollUsernameText).toBe('By: You')
  })

  it('the poll mode has the correct text', () => {
    const pollModeText = cmp.find('#poll-mode').text()
    expect(pollModeText).toBe('Mode:')
  })

  it('the multiple fires a update event when has clicked', () => {
    cmp.setProps({ multiple: true })

    const pollTypeButton = cmp.find('#poll-type')
    pollTypeButton.trigger('click')

    expect(cmp.emitted('update:multiple')).toBeTruthy()
    expect(cmp.emitted('update:multiple').length).toBe(1)
    expect(cmp.emitted('update:multiple')[0]).toEqual([false])
  })

  it('SlackList are inserted in the component', () => {
    const slackSections = cmp.findAll(SlackSection)
    expect(slackSections.length).toBe(2)
  })
})