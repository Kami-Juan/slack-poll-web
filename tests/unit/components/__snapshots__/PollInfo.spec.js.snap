// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PollInfo.vue should be render 1`] = `
<div
  id="poll-info"
>
  <div
    id="bot-avatar"
  />
   
  <div
    id="bot-message"
  >
    <h4
      id="bot-name"
    >
      Yellow Poll
    </h4>
     
    <div
      class="slack-section"
    >
      <div
        class="separator"
      />
       
      <div
        class="slack-slot"
      >
        <div
          id="info-group"
        >
          <h4
            id="poll-title"
          >
            I"m title
          </h4>
           
          <div
            id="poll-username"
          >
            By: You
          </div>
           
          <div>
            <span
              id="poll-mode"
            >
              Mode:
            </span>
             
            <a
              id="poll-type"
            >
              
            Single
          
            </a>
          </div>
        </div>
      </div>
    </div>
     
    <div
      id="options-slot"
    >
      <div
        class="slack-section"
      >
        <div
          class="separator"
        />
         
        <div
          class="slack-slot"
        />
      </div>
    </div>
  </div>
</div>
`;
